onnxruntime/core/optimizer/graph_transformer_factory.h

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/optimizer/graph_transformer.h"
+#include "core/optimizer/conv_activation_fusion.h"
+#include "core/optimizer/conv_add_fusion.h"
+#include "core/optimizer/identity_elimination.h"
+#include "core/optimizer/slice_elimination.h"
+#include "core/optimizer/unsqueeze_elimination.h"
+#include "core/optimizer/conv_bn_fusion.h"
+#include "core/optimizer/conv_mul_fusion.h"
+#include "core/optimizer/gemm_activation_fusion.h"
+#include "core/optimizer/matmul_add_fusion.h"
+#include "core/optimizer/insert_cast_transformer.h"
+
+using namespace ::onnxruntime::common;
+
+namespace onnxruntime {
+/*
+* Owns all the graph transformers.
+*/
+class GraphTransformerFactory {
+ public:
+  explicit GraphTransformerFactory() {
+    InitFactory();
+  }
+
+  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(GraphTransformerFactory);
+
+  std::vector<GraphTransformer*> GetTransformers(TransformerLevel level) const {
+    std::vector<GraphTransformer*> transformers;
+    const auto entry = level_to_string_.find(level);
+    if (entry != level_to_string_.end()) {
+      for (const auto& id : entry->second) {
+        const auto transformer = GetTransformers(id);
+        if (transformer != nullptr) {
+          transformers.push_back(transformer);
+        }
+      }
+    }
+    return transformers;
+  }
+
+  GraphTransformer* GetTransformers(std::string id) const {
+    const auto& entry = transformers_map_.find(id);
+    if (entry != transformers_map_.end()) {
+      return entry->second.get();
+    }
+
+    return nullptr;
+  }
+
+  std::vector<GraphTransformer*> GetTransformers(const std::vector<std::string>& ids) const {
+    std::vector<GraphTransformer*> transformers;
+    for (const auto& id : ids) {
+      const auto transformer = GetTransformers(id);
+      if (transformer != nullptr) {
+        transformers.push_back(transformer);
+      }
+    }
+    return transformers;
+  }
+
+  Status Register(std::unique_ptr<GraphTransformer> transformer) {
+    auto name = transformer->Name();
```

Comment:
>auto [](start = 4, length = 4)

Please be explicit and use 'const', '&' and '*' with auto. In this case, these could probably be 'const auto&'